src/common/io-config/src/python.rs

```diff
@@ -36,6 +37,29 @@ use crate::config;
 pub struct S3Config {
     pub config: crate::S3Config,
 }
+
+/// Create credentials to be used when accessing an S3-compatible system
+///
+/// Args:
+///     key_id: AWS Access Key ID, defaults to auto-detection from the current environment
+///     access_key: AWS Secret Access Key, defaults to auto-detection from the current environment
+///     session_token: AWS Session Token, required only if `key_id` and `access_key` are temporary credentials
+///     expiry: Expiry time of the credentials as a Unix timestamp in integer seconds, credentials are assumed to be permanent if not provided
+///
+/// Example:
+///     >>> get_credentials = lambda: S3Credentials(
+///     ...     key_id="xxx",
+///     ...     access_key="xxx",
+///     ...     expiry=int(time.time()) + 3600
+///     ... )
+///     >>> io_config = IOConfig(s3=S3Config(credentials_provider=get_credentials))
+///     >>> daft.read_parquet("s3://some-path", io_config=io_config)
+#[derive(Clone)]
+#[pyclass]
+pub struct S3Credentials {
```

Comment:
You can proabably leverage `#[cfg_attr(feature = "python", pyclass(module = "daft.daft"))]` instead on `S3Credentials ` which will declare it as a pyclass when the python feature is turned on.

Reply:
How would this be different from `#[pyclass]` in this situation? To my knowledge the entire file is already gated behind the python feature.